### PR TITLE
[java] Update rule to prevent UnusedImport when using JavaDoc with array type

### LINKED ISSUE #1720: [java] UnusedImports false positive for Javadoc link with array type
**Affects PMD Version:** 
Confirmed on 6.12.0 and 6.8.0

**Rule:**
UnusedImports

**Description:**

In case a JavaDoc link refers to an array typed imported type, the import is flagged as redundant.

**Code Sample demonstrating the issue:**
`FileAttribute` will be flagged as redundant.
```
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.attribute.FileAttribute; 
/**
 * See {@link Files#createDirectory(Path, FileAttribute[])}
 */
public class Test {}
```

**Running PMD through:**
Gradle/CLI

### PR DESCRIPTION
**Pull request by itaigilo**
This fixes #1720.

Allowing `[]` at the variable names of JavaDoc functions, when checking for unused imports.

### CODE REVIEW

**Comment by pmd-test** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="true">This changeset introduces 0 new violations, 0 new errors and 0 new configuration errors,<br />
removes 1 violations, 2 errors and 2 configuration errors.<br />
<a href="http://chunk.io/pmd/bf6c48017fae4285b5a9fe355cb4332c/diff/index.html">Full report</a></td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/bestpractices/UnusedImportsRule.java [hunk 1]
```diff
@@ -44,7 +44,7 @@
             .compile("@see\\s+(\\p{Alpha}?+\\w*)(?:#\\w*(?:\\(([\\w\\s,]*)\\))?)?");
```
Comment (adangel) [inline pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/bestpractices/UnusedImportsRule.java]: Wouldn't we run into the same problem with `@see SomeClass#method(MyArray[])`?
Could you please add a test for this as well?

Comment (itaigilo) [reply pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/bestpractices/UnusedImportsRule.java]: Sure, this totally makes sense.

**Comment by itaigilo** [comment]
@adangel - Updated according to your suggestion.

**Comment by adangel** [comment]
Thanks for your contribution! It will be part of PMD 6.14.0 :tada:

**Pull request merged**